scripts/build/build.js

```diff
@@ -59,28 +61,59 @@ for (const file of declarationFiles) {
   cpSync(file, ctsFile);
 }
 
-const builds = /** @type {const} */ ([
+/** @type {Array<{ format: import("tsup").Format, outDir: string, options?: Record<string, any> }>} */
+const builds = [
   { format: "esm", outDir: esmDir },
   { format: "cjs", outDir: cjsDir },
-]);
+  {
+    format: "esm",
+    outDir: solidSourceDir,
+    options: {
+      skip: !isSolid,
+      outJsExtension: ".jsx",
+      solidSource: true,
+    },
+  },
+];
 
 await Promise.all(
-  builds.map(({ format, outDir }) =>
-    build({
-      entry,
+  builds.map(
+    ({
       format,
       outDir,
-      // dts: true,
-      // tsconfig: "tsconfig.build.json",
-      splitting: true,
-      esbuildOptions(options) {
-        options.chunkNames = "__chunks/[hash]";
-        if (format === "esm") {
-          options.banner = {
-            js: '"use client";',
-          };
-        }
-      },
-    }),
+      options: {
+        skip = false,
+        outJsExtension = ".js",
+        solidSource = false,
+      } = {},
+    }) =>
+      !skip &&
+      build({
+        entry,
+        format,
+        outDir,
+        // dts: true,
+        // tsconfig: "tsconfig.build.json",
+        splitting: true,
+        esbuildOptions(options) {
+          options.chunkNames = "__chunks/[hash]";
+          if ((isCore || isReact) && format === "esm") {
+            options.banner = {
+              js: '"use client";',
+            };
+          }
+          if (isSolid) {
+            options.jsx = "preserve";
+          }
+        },
+        outExtension() {
+          return { js: outJsExtension };
+        },
+        esbuildPlugins: [
+          ...(isSolid && !solidSource
+            ? [solidPlugin({ solid: { generate: "dom" } })]
+            : []),
+        ],
+      }),
```

Comment:
It seems there are enough branches in this code to justify moving the solid build logic out of the `builds` array/map.

Something like:
```jsx
await Promise.all(
  builds.map(...),
  isSolid && build(...
```

Reply:
True. Will do that!